examples/models/ResNet50.cpp

```diff
@@ -0,0 +1,1293 @@
+    auto pool5_dst_memory = memory(pool5_dst_md, eng);
+    do_pool(eng, net, net_args, pool5_strides_dims, pool5_kernel_dims,
+            pool5_padding_dims_l, pool5_padding_dims_r, res5c_add_dst_memory,
+            pool5_dst_memory, algorithm::pooling_avg_exclude_padding);
+
+    /* -----------------inner product----------------- */
+    IC = 2048, OC = 1000;
+    memory::dims ip_src_dims = pool5_dst_md.dims();
+    memory::dims ip_weights_dims = {OC, IC, 1, 1};
+    memory::dims ip_bias_dims = {OC};
+    memory::dims ip_dst_dims = {BATCH, OC};
+
+    std::vector<char> ip_weights_data(product(ip_weights_dims) * sizeof(float));
+    std::vector<char> ip_bias_data(product(ip_bias_dims) * sizeof(float));
+
+    auto ip_bias_md = memory::desc(ip_bias_dims, dt::f32, tag::a);
+    auto ip_bias_mem = memory(ip_bias_md, eng);
+    auto ip_weights_mem = memory({ip_weights_dims, dt::f32, tag::hwio}, eng);
+    load_weights_fc(1000, ip_weights_data.data(), product(ip_weights_dims),
+            ip_bias_data.data(), product(ip_bias_dims));
+    write_to_dnnl_memory(ip_bias_data.data(), ip_bias_mem);
+    write_to_dnnl_memory(ip_weights_data.data(), ip_weights_mem);
+
+    auto ip_dst_md = memory::desc(ip_dst_dims, dt::f32, tag::nc);
+    auto ip_dst_mem = memory(ip_dst_md, eng);
+    do_ip(eng, net, net_args, ip_weights_mem, ip_bias_mem, pool5_dst_memory,
+            ip_dst_mem, true);
+    /* --------- softmax --------- */
+    const int axis = 1;
+    auto softmax_d = softmax_forward::desc(
+            prop_kind::forward_inference, ip_dst_mem.get_desc(), axis);
+    auto softmax_pd = softmax_forward::primitive_desc(softmax_d, eng);
+    auto softmax_prim = softmax_forward(softmax_pd);
+
+    net.push_back(softmax_prim);
+    net_args.push_back(
+            {{DNNL_ARG_SRC, ip_dst_mem}, {DNNL_ARG_DST, ip_dst_mem}});
+    /* ------------------------------------------ */
+    std::cout << "Warming up... BATCH SIZE = " << BATCH << "\n";
+
+    assert(net.size() == net_args.size() && "something is missing");
+    for (size_t j = 0; j < 5; j++) {
+
+        for (size_t i = 0; i < net.size(); ++i) {
+
+            net.at(i).execute(s, net_args.at(i));
+            // auto dst_mem = net_args.at(i).at(DNNL_ARG_DST);
+            // std::vector<float> dst_data(product(dst_mem.get_desc().dims()));
+            // read_from_dnnl_memory(dst_data.data(), dst_mem);
+            // check_batch(dst_data, BATCH);
+        }
+    }
+    s.wait();
+    std::cout << "Benchmarking now...\n";
+    // Execute model
+    //std::cout << "\n###########################################################"
+    //             "#############################################################"
+    //             "#############################################################"
+    //             "#########\n";
+
+    auto begin = std::chrono::high_resolution_clock::now();
+    for (int j = 0; j < times; ++j) {
+        // assert(net.size() == net_args.size() && "something is missing");
```

Comment:
This assert can be removed as it is done above.

Reply:
True, removed.